src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -409,7 +409,15 @@ ostree_repo_static_delta_execute_offline_with_signature (OstreeRepo *self, GFile
         return glnx_throw_errno_prefix (error, "openat(O_DIRECTORY)");
       else
         {
-          g_autofree char *dir = dirname (g_strdup (dir_or_file_path));
+          g_autofree char *dir = "";
+          if (dir_or_file_path[strlen (dir_or_file_path) - 1] == G_DIR_SEPARATOR)
+            {
+              g_autofree char *modifed_dir = g_strdup (dir_or_file_path);
+              modifed_dir[strlen (modifed_dir) - 1] = '\0';
+              dir = g_path_get_dirname (modifed_dir);
```

Comment:
BTW you say in your commit message: 
> We use `g_path_get_dirname()` which modifies the string in place, to which coverity should understand.

But AFACIS that's wrong, it's actually `dirname` that modifies in place, not `g_path_get_dirname`.